File: 1389-create-target-array-in-the-given-order/1389-create-target-array-in-the-given-order.c


/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* createTargetArray(int* nums, int numsSize, int* index, int indexSize, int* returnSize){

    *returnSize = numsSize;
    
    int *arr = (int*)malloc(sizeof(int)*numsSize);
    
    for (int i = 0; i < indexSize; i++){
         if(index[i] < indexSize - 1){
             for ( int  j = indexSize -1 ; j > index[i] ; j--){
                 arr[j] = arr[j-1];
             }
         }
        arr[index[i]] = nums[i];
    }
    return arr;
}

File: 1389-create-target-array-in-the-given-order/1389-create-target-array-in-the-given-order.java
class Solution {
    public int[] createTargetArray(int[] nums, int[] index) {
        List<Integer> mylist = new ArrayList<>();
        int[] arr = new int[nums.length];
        for (int i = 0; i < nums.length;i++){
            mylist.add(index[i],nums[i]);
        }
        for (int i = 0; i < arr.length;i++){
            arr[i] = mylist.get(i);
        }
        return arr;   
    }
}